### ISSUE TITLE
2.4.3 Focus Order: Should backward focus order be equivalent to forward focus?

### ISSUE THREAD

**Issue by A11yTea** [w3c/wcag#4191]
Hi everyone,

This is "kimviens" I got locked out of my old github account.

So I thought that the obvious answer to "Should backward focus order be equivalent to forward focus?" was "Yes, always." but nowhere in the understanding document does it hint at this. Not even the techniques. 

Let's say that the forward focus is (using Tab key only):
1. Button Search
2. Search text box
3. Grouping Filters, Filter by number button
4. Filter by preference button
5. Filter by date button
6. Show additional filters expandable button

But then the backward focus is (using Tab key only):
1. Show additional filters expandable button
2. Grouping Filters, Filter by number button
3. Search text box
4. Button Search

If the backward focus skipped some items however the forward focus is okay (and this is not a dialog, I repeat this is not a dialog). Does it pass 2.4.3 Focus Order?

**Comment by patrickhlauke** [comment]
yes, direction is irrelevant - even though the definition for "navigated sequentially" https://www.w3.org/TR/WCAG22/#dfn-navigated-sequentially talks of "advancing" and "next", i'd take these to mean "in the 'direction' that you're navigating", so if you're navigating in reverse, mentally flip those terms

**Comment by patrickhlauke** [comment]
so if focus order is only logical/correct going one direction, but not the other, then i would fail the latter under 2.4.3

**Comment by A11yTea** [comment]
Excellent, I agree with your argument. Thank you Patrick Lauke, this issue is closed unless someone asks to reopen it.

**Comment by detlevhfischer** [comment]
@patrickhlauke 
> so if focus order is only logical/correct going one direction, but not the other, then i would fail the latter under 2.4.3

There are cases where the tab order forward and back is different and it seems OK from a practical perspective. I'm thinking of a navigation menu design where tabbing beyond a top level menu item opens the respective sub-menu popup, continues after sub-menu items with the next top level menu item, opens the next sub-menu, and so on. Tabbing backwards then just runs through toplevel menu items, skipping the closed sub-menus (which you would enter tabbing forward). This probably no an ideal pattern, but one that I think is an OK option if used consistently -- and I would not fail this against 2.4.3.The advantage of the pattern is that tabbing back, you reach the beginning of the page (or earlier top level menu items) more quickly.

**Comment by patrickhlauke** [comment]
@detlevhfischer oh of course, there will be nuance. not saying that the focus order *needs to necessarily be exactly the same (just in reverse), but it needs to still be logical/sensible - from the example provided in the thread starter, that seemed a case where things were completely skipped in an illogical manner
